org.eclipse.xtext/src/org/eclipse/xtext/linking/lazy/LazyLinkingResource.java

```diff
@@ -120,11 +120,24 @@ protected void doLinking() {
 		if (isEagerLinking())
 			EcoreUtil.resolveAll(this);
 	}
-
+	
+	/**
+	 * The method can be overridden to avoid creation of a Set object for resources loaded from storage 
+	 * or when no cyclic resolution takes place.
+	 * @since 2.38
```

Comment:
Unfortunately it collided with the release. Should be 2.39 :/

Reply:
I understand. I have changed it to 2.39.